pandas/core/arrays/integer.py

```diff
@@ -345,6 +346,51 @@ def __array__(self, dtype=None):
         """
         return self._coerce_to_ndarray()
 
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # For IntegerArray inputs, we apply the ufunc to ._data
+        # and mask the result.
+        out = kwargs.get('out', ())
+
+        for x in inputs + out:
+            if not isinstance(x, self._HANDLED_TYPES + (IntegerArray,)):
+                return NotImplemented
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not NotImplemented:
+            return result
+
+        mask = np.zeros(len(self), dtype=bool)
+        inputs2 = []
+        for x in inputs:
+            if isinstance(x, IntegerArray):
+                mask |= x._mask
+                inputs2.append(x._data)
+            else:
+                inputs2.append(x)
+
+        def reconstruct(x):
+            if np.isscalar(x):
+                # reductions.
+                if mask.any():
+                    return np.nan
+                return x
+            if is_integer_dtype(x.dtype):
```

Comment:
how does this handle an upcasted result, e.g. we do ``np.exp(pd.Series([1,2,3], dtype='Int64'))``? and this is now a float array, does it hit this machinery?

Reply:
Yeah, it returns a float64 series (this is tested).